src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -613,6 +613,12 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
 
                 return;
             }
+            catch
+            {
+                // always dispose cancellation token registration
+                cancellationRegistration.Dispose();
```

Comment:
To simplify all this, shouldn't we be able to just do using above when calling StartCancellableOperation? We still need to manually Dispose before recursively calling ourselves - that that means we'd call Dispose twice for that case - but that should be fine no?

Reply:
Yeah, we can do that, disposing `CancellationTokenRegistration` multiple times is fine.